NWaves/Audio/WaveFile.cs

```diff
@@ -122,13 +122,25 @@ protected void ReadWaveStream(Stream waveStream, bool normalized = true)
                 waveFmt.Align = reader.ReadInt16();
                 waveFmt.BitsPerSample = reader.ReadInt16();
 
-                WaveFmt = waveFmt;
-
-                if (fmtSize == 18)
+                // Header size might not include sizeof extension block.
+                if (fmtSize == 18 || fmtSize == 40)
                 {
                     var fmtExtraSize = reader.ReadInt16();
-                    reader.ReadBytes(fmtExtraSize);
+                    // Additional info for the following:  https://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/WAVE.html
```

Comment:
Looks like a good addition to the library. Could you delete comments on lines 129-130 and 136?